packages/@storefront/flux-capacitor/src/core/reducers/data/navigations.ts

```diff
@@ -65,21 +67,30 @@ export const receiveNavigations = (state: State, navigations: Store.Navigation[]
 };
 
 // tslint:disable-next-line max-line-length
-export const selectRefinement = (state: State, { navigationId, index: refinementIndex }: Actions.Payload.Navigation.Refinement) => {
-  if (navigationId && refinementIndex != null) {
-    return {
-      ...state,
-      byId: {
-        ...state.byId,
-        [navigationId]: {
-          ...state.byId[navigationId],
-          selected: state.byId[navigationId].selected.concat(refinementIndex),
-        },
+export const selectMultipleRefinements = (state: State, { navigationId, indices }: Actions.Payload.Navigation.MultipleRefinements) => {
+  if (!navigationId || indices == null) {
```

Comment:
What are you really trying to check with `indices == null` here? Indices is an array, and you have a validator in the action that reaches this reducer to verify that it sends `indices` as an array, then in the `selectRefinement` function that calls this function, it puts its only value into an array. In other words, this will always be an array, so `indices == null` will never be true. If you want to still be _safe_ I guess you can keep it in, just wanted to point this out

Reply:
I guess what I'm trying to say is a check like this probably belongs in the action before it reaches this code. Maybe you want to check if `indices` has length instead because it's a little strange to use `selectMultipleRefinements` to deselect all refinements? I'm not sure

Reply:
We were porting over the functionality from selectRefinement which was a mistake. We are now checking against `indices.length === 0`. In terms of adding checks to the actions, we already have our validators that are associated with the actions, so would we need the extra check of including to see if indices has a length in that spot?